### DISCUSSION TITLE
Comments on commit not visible on iOS app

### DISCUSSION THREAD

**Discussion by konacurrents** [community/community#28069] [Mobile] [answered]
When someone comments on a commit, I get an email with the contents. Replying via email works and adds to the comment thread. 

Unfortunately clicking on the `view it on GitHub` just shows the code changes. 

There is no link to the comments. From the app I cannot see them in all my various attempts. 

The `issues` works great on app, although floating down the Amazon without internet might make that hard (as my wife just did). 

Thanks.

**Comment by stevepopovich** [comment]
Hi @konacurrents. So am I correct and understanding that your problem is that the `view it on Github` does not deeplink to a reply from an email? Is it deeplinking into the mobile app?

**Reply by konacurrents** [reply]
So, yes the view-on-github is taking me to the mobile app. 

But the mobile app just shows me the file diff (1 file changes, 1 addition, 1 deletion .. etc). 

But there is nothing below the diff showing all the comments. I notice those show up on the web version.
![IMG_8624](https://user-images.githubusercontent.com/5956266/182459199-5f03d90a-2b05-4ab9-84bc-62242ba24fbb.PNG)

**Reply by stevepopovich** [reply]
Gotcha, makes sense. I will file a bug!

**Reply by konacurrents** [reply]
@stevepopovich update:
Now the email I get which has an `open on GitHub` gives me a 404 error instead of taking me to the app. 

Link in email: [https://github.com/konacurrents/ESP_IOT/issues/157#issuecomment-1226102292](https://github.com/konacurrents/ESP_IOT/issues/157#issuecomment-1226102292)

Note: the parsing of this http link doesn't show correct either, maybe the # breaks syntax?

![image](https://user-images.githubusercontent.com/5956266/186501968-2ec2b141-3ef6-44ed-b56a-4d3dec06a0bd.png)

**Reply by stevepopovich** [reply]
Gotcha, ok, updated our internal issue

**Reply by alialhammad1992** [reply]
> Gotcha, ok, updated our internal issue

 The same problem appears with me iPhone 13 Pro Max

**Reply by konacurrents** [reply]
Any updates? 

Each email I get with a link goes to a 404 not found link. 
(And I get an email for every issue update, all not working). 

They were working before I started the issue earlier.

**Reply by K-bit1** [reply]
And what can i do to be found in 404

**Comment by konacurrents** [comment]
Note, this issue on the mobile app you just sent does NOT take me to the app, but instead to the web page. (And since it's my mobile device, that web site doesn't know about me.) I was going to add an image..

Here is the image I just mentioned.
![IMG_8625](https://user-images.githubusercontent.com/5956266/182459118-59324d98-c8f9-4a39-bc38-243ff770c09d.PNG)

**Comment by konacurrents** [comment]
As I mentioned, the `view on GitHub `doesn't even work now (after I first raised this issue). This is for `issues`.

I don't know what state this thread is but there are no answers I've seen

thanks,

**Comment by konacurrents** [comment]
Ok more info:
From my iPhone email tool, the link brings up the 404 error saying I have to log in.
But from my google email on my mac, the same link works fine. This is because my web browser is logged into GitHub. (Probably doesn't work if not).

### The issue if access rights

Using this [link](https://github.com/orgs/community/discussions/28069#discussioncomment-3552232) to this issue as an example, the `community` is open to all - so the iPhone at least brings up that issue in a web browser. But NOT in the GitHub app. The web browser asks me to sign in or sign up.

But using a link to my private GitHub, that same web browser brings up 404 since it doesn't know me, and access rights say I cannot see that issue.

**Comment by stevepopovich** [comment]
Sorry, @konacurrents. What is going on seems to have evolved. You seem to have went from asking for better deeplink support from emails to seeing 404s. Can you clarify what issues you are having?

**Reply by konacurrents** [reply]
Steve, I don't know what the comments from @K780420 are coming from. 
My issue is straight forward and has to do with permissions and app interaction on the iPhone. The evolution is in the GitHub app since the behavior changed since I started this issue.

### Open email on iPhone (link doesn't work for private repository)
1. The email I receive on issues has a link to 'view on GitHub'. The email wants to open my web browser (safari) with that link. It does NOT try to open the GitHub app.
2. Since it's trying to open a web browser, on the iPhone, there are no login credentials for the web browser (since I use the GitHub app, not the browser).
3. So depending on the permissions of the link (a public or private repository), the results of the web browser are different.
4. If a public repository, the web browser brings up the correct issue, but asks me to log into my account (or sign up)
5. If a private repository, the web browser shows a 404 (link not found) page.

### Open email on Mac using safari (not mac's email)

1. My email browser is gmail so it's a web interface, on one of my safari tabs. 
2. In another  safari tab I have GitHub  logged in.
3. Clicking on the link in gmail, 'view on GitHub', correctly opens a new tab for GitHub and takes me to the correct issue. Permissions are ok since I've already logged-in in another tab.

Hope that clarifies the issue. At present, the email on iPhone link is useless.
thanks.

**Reply by stevepopovich** [reply]
Gotcha. Thank you very much. It sounds like you just need to log in on the web browser and things should start working. I would expect private repos you don't have access to 404. If you are logged out, you have access to no private repos.

**Reply by konacurrents** [reply]
What I mentioned is that on the iPhone, there isn't a good integration with the GitHub app. The link brings up a web browser and since I use the GitHub app - the web version is NOT used. So logging in isn't a viable option. 

### Instead I get this message if I click on a public repository link:
<img width=200 src="https://user-images.githubusercontent.com/5956266/190485333-3d43fe49-7ae5-4d87-9521-f2fdaa5206df.jpg">

###  And Private repository shows this:
<img width=200 src="https://user-images.githubusercontent.com/5956266/190485649-6584d553-3fec-4ebb-9256-d66a03c37876.jpg">

**Reply by stevepopovich** [reply]
Gotcha. Why can't you hit `sign in` or `Sign in to comment`? You should be able to have a log in instance in the browser and be logged in the mobile app the same time. This is a general flow for users, since the mobile doesn't have every feature that the website has.

**Reply by konacurrents** [reply]
I have to go through 2 factor authentication, etc. I don't use the web browser on the iPhone for GitHub. 

What I don't get, is this was working when I started this issue - but something in the GitHub code changed to make bringing up the GitHub app not working.

**Reply by stevepopovich** [reply]
Are you not able to use 2FA through the mobile web?

Yeah. I am not sure either. We shipped 2FA for mobile many months ago, that may have changed things. But it does sound that things are working for you as we intend, just might not be the best experience for you 😞

**Reply by konacurrents** [reply]
I guess I'll stop on this thread. The experience for the iPhone user is unacceptable. I never use the web version of GitHub on the iPhone - so logging in via 2FA is not appropriate for the mobile user. That's what the GitHub app is for.

If there is no way for the email link to bring up the GitHub app then that is too bad, and sadly not usable.

### zoom app
Take the popular zoom app. In my email or a calendar event, is a link to the zoom meeting. On the iPhone, clicking on that will bring up the zoom app (not a web page that mimics zoom.) The actual zoom app.

That is what GitHub should strive for in their mobile implementation: and email link opens up the GitHub mobile app at the correct point in whatever issue is linked.

thanks..

**Reply by stevepopovich** [reply]
Gotcha. Yes, I understand the original deeplink problem. I have logged your feature request and maybe it something will we add in the future.

**Reply by konacurrents** [reply]
Thanks, that would make your product very powerful.
cheers..

**Reply by cartpauj** [reply, marked as answer]
+1 to please add commit comments to Github mobile app. (iOS)

**Comment by stevepopovich** [comment]
@K780420 Please hit this `Unsubscribe` button to stop getting tagged in this thread: 
<img width="352" alt="image" src="https://user-images.githubusercontent.com/3252375/190509194-876fb6a3-432e-4c2d-81ed-6a0a007861fe.png">
